### PR TITLE
Add remarks to FromBodyAttrbiute

### PR DESCRIPTION
**Pull request by voroninp**
# Add remarks to FromBodyAttrbiute

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

[[FromBody] attrbiute](https://learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-8.0#frombody-attribute) section of the docs dedicated to Model Binding contains a very important statement:
> The ASP.NET Core runtime delegates the responsibility of reading the body to an input formatter.

This is quite an important detail (gotcha, actually) to capture it in the xmldoc of the attribute, so it's imediately visible in IDE.

### CODE REVIEW

File: src/Mvc/Mvc.Core/src/FromBodyAttribute.cs [hunk 1]
```diff
@@ -9,6 +9,9 @@ namespace Microsoft.AspNetCore.Mvc;
 /// <summary>
 /// Specifies that a parameter or property should be bound using the request body.
 /// </summary>
+/// <remarks>
+/// Be default ASP.NET Core runtime delegates the responsibility of reading the body to an input formatter.
```
Comment (joegoldman2) [inline src/Mvc/Mvc.Core/src/FromBodyAttribute.cs:13]: This is true for MVC but not for minimal API, which also uses the same attribute.

Comment (voroninp) [reply src/Mvc/Mvc.Core/src/FromBodyAttribute.cs:13]: How does it behave for minipal API then?

Comment (martincostello) [reply src/Mvc/Mvc.Core/src/FromBodyAttribute.cs:13]: I think it just reads the form itself directly before passing control to the user's endpoint method using https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.http.features.iformfeature.readformasync (I haven't dug through the code to check, but that's what I see to remember from doing work on the binding "glue" stuff).

Comment (voroninp) [reply src/Mvc/Mvc.Core/src/FromBodyAttribute.cs:13]: Found it:
[Here](https://github.com/dotnet/aspnetcore/blob/98ee50279acfacff2b4ae4b6863956f3c65f34b2/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L735) and then [here](https://github.com/dotnet/aspnetcore/blob/98ee50279acfacff2b4ae4b6863956f3c65f34b2/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L1390).

Ok, I need to change the wording. The fun thing that MVC's model binding involves validation, and even System.Text.Json does not care about `[Required]` attrbiute I get 400 Bad Request while for minimal API I get 200.

Comment (voroninp) [reply src/Mvc/Mvc.Core/src/FromBodyAttribute.cs]: @martincostello , I changed the wording.

File: src/Mvc/Mvc.Core/src/FromBodyAttribute.cs [hunk 2]
```diff
@@ -9,6 +9,10 @@ namespace Microsoft.AspNetCore.Mvc;
 /// <summary>
 /// Specifies that a parameter or property should be bound using the request body.
 /// </summary>
+/// <remarks>
+/// Be default for MVC ASP.NET Core runtime delegates the responsibility of reading the body to an input formatter.<br/>
+/// In the case of Minimal API, the body is deserialized by <see cref="System.Text.Json.JsonSerializer">JsonSerializer</see>.
```
Comment (martincostello) [inline src/Mvc/Mvc.Core/src/FromBodyAttribute.cs:14]:
```suggestion
/// By default, ASP.NET Core MVC delegates the responsibility of reading the body to an input formatter.<br/>
/// In the case of ASP.NET Core Minimal APIs, the body is deserialized by <see cref="System.Text.Json.JsonSerializer"/>.
```

**Comment by KennethHoff** [comment]
What exactly is "an input formatter"?

**Comment by voroninp** [comment]
@KennethHoff ,is it a hint that link to the description of formatters concept is missing or a real question?

If it's the latter one , then
https://learn.microsoft.com/en-us/aspnet/core/web-api/advanced/formatting?view=aspnetcore-8.0

**Comment by KennethHoff** [comment]
> @KennethHoff ,is it a hint that link to the description of formatters concept is missing or a real question?

Both, really 😅

**Review by captainsafia — APPROVED** [review]
LGTM!

Out of curiousity, what motiviated you to add this note? It seems like you might not have expected input formatters to be coming in to play here?

**Comment by voroninp** [comment]
@captainsafia , well, I knew this, but I've learned it the hard way by making wrong assumptions. And minimal API behavior was a surprise.
This time I've noticed my colleague had wrong expectations about model validation.

**Comment by voroninp** [comment]
@captainsafia , actually, there are many places where xmldocs are lacking important details. Sometimes, clarifications can be found in docs at Microsoft.Learn. But quite often I need to look into sources.

For [exception handler middleware](https://github.com/dotnet/aspnetcore/blob/3f1acb59718cadf111a0a796681e3d3509bb3381/src%2FMiddleware%2FDiagnostics%2Fsrc%2FExceptionHandler%2FExceptionHandlerExtensions.cs) it is not mentioned that by default 404 status code without started response will cause `InvalidOperationException`.

I do not get the description for [this overload](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.exceptionhandlerextensions.useexceptionhandler?view=aspnetcore-8.0#microsoft-aspnetcore-builder-exceptionhandlerextensions-useexceptionhandler(microsoft-aspnetcore-builder-iapplicationbuilder)).
I do not specify any url, what is an alternative pipeline in this case, what for?

I'd also change the wording of the comment for [adding a custom exception handler](https://source.dot.net/#Microsoft.AspNetCore.Diagnostics/ExceptionHandler/ExceptionHandlerServiceCollectionExtensions.cs,e74aac24e3e2cbc9) from "it is used by exception handler middleware" to "it is **required** to use exception handler middleware for the custom handler to work. (Analyzer would be even better;-) )

For authentication it's not obvious when `Thread.CurrentPrincipal` is set and when it is not.

Why is [`Thread.CurrentPrincipal`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.thread.currentprincipal?view=net-8.0) nullable, but [`HttpContext.User`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httpcontext.user?view=aspnetcore-8.0) is not?

**Pull request merged**